server/jetstream_benchmark_test.go

```diff
@@ -1295,6 +1295,187 @@ func BenchmarkJetStreamObjStore(b *testing.B) {
 	}
 }
 
+func BenchmarkJetStreamFanIn(b *testing.B) {
+	const (
+		subject     = "test-subject"
+		streamName  = "test-stream"
+		clusterSize = 3
+		replicas    = 3
+	)
+
+	type BenchPublisher struct {
+		// nats connection for this publisher
+		conn *nats.Conn
+		// jetstream context
+		js nats.JetStreamContext
+		// number of publishing errors encountered
+		publishErrors int
+		// quit channel which will terminate publishing
+		quitCh chan bool
+	}
+
+	messageSizeCases := []int64{
+		100,        // 100B
+		1024,       // 1KiB
+		10240,      // 10KiB
+		512 * 1024, // 512KiB
+	}
+	numPubsCases := []int{
+		3,
+		5,
+		10,
+	}
+
+	fanIn := func(b *testing.B, numPubs int, messageSize int64) {
+
+		cl, ls, shutdown, nc, js := startJSClusterAndConnect(b, clusterSize)
```

Comment:
I'd leave cluster creation (maybe stream too) down in the "main" loop, and just have benchmark code here in this benchmark function.